src/hooks/LogHookContext.ts

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License.
+
+import { HookData, RpcLogCategory, RpcLogLevel } from '@azure/functions-core';
+import * as coreTypes from '@azure/functions-core';
+import { AzureFunctionsRpcMessages as rpc } from '../../azure-functions-language-worker-protobuf/src/rpc';
+import { toCoreLog } from '../coreApi/converters/toCoreStatusResult';
+import { ReadOnlyError } from '../errors';
+import { nonNullProp } from '../utils/nonNull';
+import { worker } from '../WorkerContext';
+
+export interface InvocationLogContext {
+    hookData: HookData;
+    invocationContext: unknown;
+}
+
+export class LogHookContext implements coreTypes.LogHookContext {
+    level: RpcLogLevel;
+    message: string;
+    #hookData: HookData;
```

Comment:
Nit: should the private fields be listed first? And should we sort them alphabetically?

Reply:
There's no functional reason we need to sort them. We could do it as a stylistic choice, but I'm not a big fan of stylistic rules unless we have a linter or formatter to enforce them. Seems like both eslint and prettier deemed this kind of sorting as too complex to be worth supporting:
- https://github.com/prettier/prettier/issues/5412
- https://github.com/typescript-eslint/typescript-eslint/issues/8792

The eslint rule is in a weird limbo state and we could still decide to use it, but it would need to be enabled in a separate PR for the whole repo. The default is private fields last, btw, which is what I'm used to.